iXBRLViewerPlugin/__init__.py

```diff
@@ -11,9 +11,33 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
+# GUI operation:
+#
+#     if submenu View->iXBRL Viewer->Show iXBRL Filing Data is checkmarked, a local viewer is automatically opened to view
+#
+#     to save a viewable file Tools->Save iXBRL Viewer Instance (dialog requests linkable js location and save location)
+#
+# Command line operation:
+#
+#     parameters --save-viewer (file system location to save at) and --viewer-url (linkable js location)
+#
+# Web Server operation:
+#
+#     example uploading an ESEF report package and receiving a zip of viewable .xhtml and viewer javascript file:
+#
+#         curl -X POST "-HContent-type: application/zip" 
+#              -T /Users/mystuff/ESMA/samples/bzwbk_2016.zip 
+#              -o ~/temp/out.zip 
+#              "http://localhost:8080/rest/xbrl/validation?&media=zip&plugins=iXBRLViewerPlugin&packages=somewhere/esef_taxonomy_2017.zip"
+#
+
 
 from .iXBRLViewer import IXBRLViewerBuilder
+import os, zipfile, sys, traceback
+from arelle.FileSource import archiveFilenameParts
 
+BUILTIN_SCRIPT_URL = os.path.join(os.path.dirname(__file__), "viewer", "dist", "ixbrlviewer.js")
```

Comment:
The root of `viewer/dist` is used in multiple places in this PR can we utilize this constant else where so that if its decided to move the ixbrlviewer.js file to a different loocation we dont have to update multiple places in the code?

Reply:
Of course